plugin/al/lib/AL_USDMaya/AL/usdmaya/nodes/TransformationMatrix.cpp

```diff
@@ -872,37 +896,44 @@ void TransformationMatrix::initialiseToPrim(bool readFromPrim, Transform* transf
     m_flags |= kInheritsTransform;
 
   if(matchesMayaProfile(m_xformops.begin(), m_xformops.end(), m_orderedOps.begin()))
-  {
     m_flags |= kFromMayaSchema;
-  }
-  else
-  {
-  }
+
+  // Disable push to prim if enabled, otherwise MPlug value queries and setting
+  // in the switch statement below will trigger pushing to the prim, which
+  // will create undesirable usd "over"s.
+  //
+  bool pushOriginallyEnabled = pushToPrimEnabled();
+  m_flags &= ~kPushToPrimEnabled;
 
   auto opIt = m_orderedOps.begin();
   for(std::vector<UsdGeomXformOp>::const_iterator it = m_xformops.begin(), e = m_xformops.end(); it != e; ++it, ++opIt)
   {
     const UsdGeomXformOp& op = *it;
+
+    // How many time samples does the op have?  (This will be 0 if the op only has a constant
+    // value set or if it has no value set at all.)
+    //
+    int numTimeSamples = op.GetNumTimeSamples();
+
     switch(*opIt)
     {
     case kTranslate:
       {
         m_flags |= kPrimHasTranslation;
-        if(op.GetNumTimeSamples() > 1)
-        {
+        if(numTimeSamples > 0)
           m_flags |= kAnimatedTranslation;
```

Comment:
I see that you're changing the condition here, from > 1 to > 0.

This makes the behavior in `initialiseToPrim` match that in `updateToTime`, which is good... but I'm wondering what your rationale was for changing everything to > 0, as opposed to making everything be > 1.  At a guess, it seems like > 1 might make more sense as a criteria for whether there's animation, since with only 1 I would assume the value would still be static.

Might also be good to hear from someone at AL on whether there was a reason for the different criteria in updateToTime vs initialisePrim, or whether that was just an oversight.  @murphyeoin  or @robthebloke ?

Reply:
Since there is an explicit way in usd to set a constant value (e.g. "float3 xformOp:translate = (1,2,3)"), and a different way is used for time samples (e.g. "float3 xformOp:translate.timeSamples = {1: (1,2,3), 2: (..."), even if there is only one time sample it seems like the setting with time samples should be considered animation, as additional time samples may be added at any time.
In trying to address the problems we were encountering (unfortunately without having a good understanding of why the code was set up how it was and whether there were really good or maybe not so good reasons for it), I had to try a variety of things.  This was one of the tweaks that made intuitive sense to me and seemed to help so I kept it.

Reply:
Hmm, well that argument makes sense to me. We’ll stick with your > 0 unless else objects...